sieve.py

```diff
@@ -325,7 +325,7 @@ def check_result(
     project, mode, stage, test_config, log_dir, data_dir, two_sided, oracle_config
 ) -> Tuple[int, str]:
     if stage == "learn":
-        analyze.analyze_trace(project, log_dir, two_sided=two_sided)
+        analyze.analyze_trace(project, log_dir, data_dir, two_sided=two_sided, learn_twice=(mode=="learn-twice"))
```

Comment:
Since we are passing `data_dir` to `analyze_trace`, can you move the following
```
        cmd_early_exit("mkdir -p %s" % data_dir)
        cmd_early_exit(
            "cp %s %s"
            % (
                os.path.join(log_dir, "xxx.json"),
                os.path.join(data_dir, "xxx.json"),
            )
        )
```
into `analyze_trace`? For example, it can reside after `if generate_oracle:`

Reply:
Note that the output of the two run should be:
1. For the first learn run, we want to get the generated test configs, the side effect digest, and the resources.json. The side effect digest should be stored at `data_dir` but others should just remain in `log_dir`
2. For the second learn run, we should only generate the resources.json. We then compare the two resources.json and move the trimmed one to `data_dir`